Core/NoOpRunner.Core/Shapes/EntityShapes/DamageCrystal.cs

```diff
@@ -0,0 +1,21 @@
+﻿using NoOpRunner.Core.Entities;
+using NoOpRunner.Core.Enums;
+
+namespace NoOpRunner.Core.Shapes.EntityShapes
+{
+    class DamageCrystal : EntityShape
+    {
+        public DamageCrystal(int centerPosX, int centerPosY) : base(centerPosX, centerPosY)
+        {
+            MapShapeY(1, 1, 2, Color.Red);
+        }
+
+        public override void OnCollision(BaseShape other)
+        {
+            if (other.GetType() == typeof(Player))
+            {
+                other.DoDamage(1);
+            }
```

Comment:
Use pattern matching. `            if (other is Player p)
            {
                // do stuff with p
            }`. Then, as you cast the shape into type Player, you won't need the `DoDamage()` function on `BaseShape` where it does not belong, you can handle player health changes in player.

Reply:
Fixed